File: CTask/src/org/ShapeSHifter420/Filters/LocationFilter.java
package org.ShapeSHifter420.Filters;

import org.ShapeSHifter420.Pushes.*;

public class LocationFilter implements IFilter{
    @Override
    public boolean isAllow(IPush push) {
        return push instanceof ILocationPush;
    }

    @Override
    public boolean test(IPush push, ISystem sys) {
        ILocationPush agepush = (ILocationPush) push;
        return agepush.getRadius() >= Math.sqrt(((Math.pow(agepush.getXCoord() - sys.getXCoord(),2))+
                (Math.pow(agepush.getYCoord() - sys.getYCoord(),2))));
    }
}

File: CKotlin/Filters/LocationFilter.kt
package Filters

import Pushes.IPush
import Pushes.ILocationPush
import Pushes.ISystemData
import java.lang.Math.sqrt
import kotlin.math.pow

class LocationFilter : IFilter {
    override fun allow(push: IPush): Boolean = push is ILocationPush

    override fun filtration(push: IPush,sysData: ISystemData): Boolean {
        val locationPush = push as ILocationPush
        return kotlin.math.sqrt(
            (sysData.xCoord - locationPush.xCoord).pow(2.0) + (sysData.yCoord - locationPush.yCoord).pow(2.0)
        ) <=locationPush.radius
    }
}